amplitude.js

```diff
@@ -801,18 +830,41 @@ AmplitudeClient.prototype._setInStorage = function _setInStorage(storage, key, v
 };
 
 /**
+ * Helper function to remove values in storage
+ * Storage argument allows for localStoraoge and sessionStoraoge
+ * @private
+ */
+AmplitudeClient.prototype._removeFromStorage = function _removeFromStorage(storage, key) {
+  storage.removeItem(key + this._storageSuffix);
+};
+
+/**
  * cookieData (deviceId, userId, optOut, sessionId, lastEventTime, eventId, identifyId, sequenceNumber)
  * can be stored in many different places (localStorage, cookie, etc).
  * Need to unify all sources into one place with a one-time upgrade/migration.
  * @private
  */
 var _upgradeCookeData = function _upgradeCookeData(scope) {
-  // skip if migration already happened
-  var cookieData = scope.cookieStorage.get(scope.options.cookieName);
+  // check if already migrated to new instance keys with scope
+  var scopedKey = scope.options.cookieName + scope._apiKeySuffix + scope._storageSuffix;
```

Comment:
Minimum 45 chars (amplitude_id_api_key, no instance name for default instance), potentially up to 60 chars depending on what kind of name they pick for their new instance. I don't think this will be an issue since browsers allow for 5-10mb for localStorage (of course this will be shared between us and the user's website, but I still think that's very generous).